Added mock, coverage and pep8 dep. Added pep8 options

montefra opened the pull request:
I've added `mock` as mandatory dependency for python < 3.3 and `coverage` and `pep8` as optional one. I've  also added the `pep8` options that I've found this morning in the `tests.py` script.

montefra commented:
why does the second call of `tests.py` is

    python ../matplotlib/tests.py -sv --processes=5 --process-timeout=300

and not 

    python tests.py -sv --processes=5 --process-timeout=300

like the first one?

pelson commented:
Thanks @montefra. Why are mock and pep8 considered mandatory dependencies? Both are 100% (optional?) test dependencies, but neither should be used in the non-test codebase.

montefra commented:
@pelson: `mock` is mandatory for testing if one uses python < 3.3, so I have added it after the `nose` entry. `coverage` and `pep8` are optional and I've added them after:

    Optionally you can install:

>  Why are mock and pep8 considered mandatory dependencies? Both are 100% (optional?) test dependencies, but neither should be used in the non-test codebase.

What do you mean? I've edited only `testing.rst`, I haven't added `mock`, `coverage` or `pep8` anywhere else.

jenshnielsen commented:
@pelson Mock is actually used to build the docs at this stage to avoid pulling in additional dependencies when running autodoc. I should probably improve the documentation to reflect that.

montefra commented:
If the commit is accepted, would it make sense to also have a message in [`setupext:Tests`](https://github.com/montefra/matplotlib/blob/cee0fa939d0314cbd60b5f6bc7f9a09eab7fe635/setupext.py#L637) if `coverage` and/or `pep8` are not installed?

pelson commented:
> What do you mean? I've edited only testing.rst, I haven't added mock, coverage or pep8 anywhere else.

My apologies, the PR title does not reflect what you have done. I'm :+1: with this change as it is.
